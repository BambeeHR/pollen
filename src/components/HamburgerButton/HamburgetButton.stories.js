import HamburgerButton from './HamburgerButton.vue';

export default {
  title: 'Buttons/HamburgetButton',
  parameters: {
    componentSubtitle: 'Hamburget menu button.',
  },
};

export const Default = () => ({
  components: { HamburgerButton },
  data() {
    return {
      isOpen: false,
    };
  },
  template: `<HamburgerButton @click="isOpen = !isOpen" :open="isOpen" />`,
});
