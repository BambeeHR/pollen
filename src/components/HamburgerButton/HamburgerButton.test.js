import { mount } from '@vue/test-utils';
import HamburgerButton from './HamburgerButton.vue';

describe('HamburgerButton', () => {
  test('renders correctly when closed', () => {
    const wrapper = mount(HamburgerButton);
    expect(wrapper.element).toMatchSnapshot();
  });
  test('renders correctly when open', () => {
    const wrapper = mount(HamburgerButton, { propsData: { open: true } });
    expect(wrapper.element).toMatchSnapshot();
  });
  test('emits a click event', () => {
    const onClick = jest.fn();
    const wrapper = mount(HamburgerButton, {
      listeners: {
        click: onClick,
      },
    });
    wrapper.trigger('click');
    expect(onClick).toHaveBeenCalled();
  });
});
