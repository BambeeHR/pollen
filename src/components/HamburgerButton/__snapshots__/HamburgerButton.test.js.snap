// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HamburgerButton renders correctly when closed 1`] = `
<button
  class="hamburger-button hamburger-button--closed"
  title="Menu"
>
  <span
    class="sr-only"
  >
    Menu
  </span>
</button>
`;

exports[`HamburgerButton renders correctly when open 1`] = `
<button
  class="hamburger-button hamburger-button--open"
  title="Menu"
>
  <span
    class="sr-only"
  >
    Menu
  </span>
</button>
`;
